ArcGISRuntimeSDKQt_CppSamples/EditData/EditWithBranchVersioning/EditWithBranchVersioning.qml

```diff
@@ -0,0 +1,348 @@
+                } else if (text === qsTr("Switch to default version")) {
+                    text = qsTr("Switch to created version")
+                    model.applyEdits();
+                    callout.dismiss();
+                    updateWindow.visible = false;
+                } else if (text === qsTr("Switch to created version")) {
+                    text = qsTr("Switch to default version")
+                    model.switchVersion();
+                    callout.dismiss();
+                    updateWindow.visible = false;
+                }
+            }
+        }
+
+        Rectangle {
+            id: currentVersionRect
+            anchors{
+                top: parent.top
+                horizontalCenter: parent.horizontalCenter
+                margins: 3
+            }
+            width: childrenRect.width;
+            height: childrenRect.height;
+            color: "#000000"
+            visible: currentVersionText.text !== "" ? true : false
+
+            ColumnLayout {
+                spacing: 3
+                Text {
+                    text: qsTr("Current version:")
+                    Layout.alignment: Qt.AlignHCenter
+                    color: "white"
+                }
+
+                Text {
+                    id: currentVersionText
+                    Layout.alignment: Qt.AlignHCenter
+                    text: model.sgdbCurrentVersion
+                    color: "white"
+                }
+            }
+        }
+    }
+
+    Rectangle {
+        id: createVersionWindow
+        anchors.centerIn: parent
+        width: childrenRect.width
+        height: childrenRect.height
+        radius: 10
+        visible: false
+
+        GaussianBlur {
+            anchors.fill: createVersionWindow
+            source: view
+            radius: 40
+            samples: 20
+        }
+
+        MouseArea {
+            anchors.fill: parent
+            onClicked: mouse.accepted = true;
+            onWheel: wheel.accepted = true;
+        }
+
+        GridLayout {
+            columns: 2
+            anchors.margins: 5
+
+            TextField {
+                id: versionNameTextField
+                placeholderText: qsTr("Name must be unique")
+                Layout.columnSpan: 2
+                Layout.alignment: Qt.AlignHCenter
+                Layout.margins: 5
+                validator: RegExpValidator { regExp: /\w{0,50}/ }
+            }
+
+            ComboBox {
+                id: accessComboBox
+                model: versionAccessModel
+                Layout.columnSpan: 2
+                Layout.fillWidth: true
+                Layout.alignment: Qt.AlignHCenter
+                Layout.margins: 5
+            }
+
+            TextField {
+                id: descriptionTextField
+                placeholderText: qsTr("Enter description")
+                Layout.columnSpan: 2
+                Layout.alignment: Qt.AlignHCenter
+                Layout.margins: 5
+            }
+
+            Row {
```

Comment:
Should this `Row` exist? It looks to me like the individual buttons should be inserted directly into the GridLayout. Otherwise the `columnSpan` of this GidLayout is superfluous, as all contents are currently 2 columns anyway.

Reply:
You know that is an excellent point! I reused this UI element from a previous sample and just hacked it up to suit this sample but now that you mention it I did it specifically for aesthetics and it would probably be better just to make it a ColumnLayout as I only want one.